src/event-manager.js

```diff
@@ -86,8 +86,8 @@ function handleDelegatedEvent(event) {
         }
       }
     }
-
-    target = target.parentNode;
+    
+    target = (target.parentNode && target.parentNode.toString() === "[object ShadowRoot]") ? target.parentNode.host : target.parentNode;
```

Comment:
Can we avoid accessing `parentNode` multiple times? It would be tough on minification, and also perf

Reply:
@bigopon Pushed a change, caching the parentNode